stats/thresholds_parser_test.go

```diff
@@ -0,0 +1,144 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2021 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package stats
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_parseThresholdExpression(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name          string
+		input         string
+		wantCondition *thresholdExpression
+		wantErr       bool
+	}{
+		{"unknown expression's operator fails", "count!20", nil, true},
+		{"unknown expression's method fails", "foo>20", nil, true},
+		{"non numerical expression's value fails", "count>abc", nil, true},
+		{"valid threshold expression syntax", "count>20", &thresholdExpression{AggregationMethod: "count", Operator: ">", Value: 20}, false},
+	}
+	for _, testCase := range tests {
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+
+			gotCondition, err := parseThresholdExpression(testCase.input)
+			if (err != nil) != testCase.wantErr {
+				t.Errorf("parse() error = %v, wantErr %v", err, testCase.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(gotCondition, testCase.wantCondition) {
+				t.Errorf("parse() = %v, want %v", gotCondition, testCase.wantCondition)
+			}
```

Comment:
Is there any reason why you are not using [testify](https://pkg.go.dev/github.com/stretchr/testify) we are after all using it through the codebase?

Reply:
Once again, I'll blame VSCode. The Go extension's `generate unit test for function` command does use the standard package only; and that's the one I use to write all the boilerplate for me. I'll modify the code to use `testify` instead, and find out if I can customize the snippet command to avoid that in the future 👍🏻